contracts/utils/HStrategyHelper.sol

```diff
@@ -0,0 +1,508 @@
+    /// @param currentTokenAmounts the amounts of tokens on the vaults
+    /// @param expectedTokenAmounts the amounts of tokens expected after rebalancing
+    /// @param ratioParams ratio of the tokens between erc20 and money vault combined with needed deviations for rebalance to be called
+    /// @return needed true if the token swap is needed
+    function swapNeeded(
+        HStrategy.TokenAmounts memory currentTokenAmounts,
+        HStrategy.TokenAmounts memory expectedTokenAmounts,
+        HStrategy.RatioParams memory ratioParams,
+        HStrategy.DomainPositionParams memory domainPositionParams
+    ) external pure returns (bool needed) {
+        uint256 expectedTotalToken0Amount = expectedTokenAmounts.erc20Token0 +
+            expectedTokenAmounts.moneyToken0 +
+            expectedTokenAmounts.uniV3Token0;
+        uint256 expectedTotalToken1Amount = expectedTokenAmounts.erc20Token1 +
+            expectedTokenAmounts.moneyToken1 +
+            expectedTokenAmounts.uniV3Token1;
+
+        uint256 currentTotalToken0Amount = currentTokenAmounts.erc20Token0 +
+            currentTokenAmounts.moneyToken0 +
+            currentTokenAmounts.uniV3Token0;
+        uint256 currentTotalToken1Amount = currentTokenAmounts.erc20Token1 +
+            currentTokenAmounts.moneyToken1 +
+            currentTokenAmounts.uniV3Token1;
+
+        int256 token0Delta = int256(currentTotalToken0Amount) - int256(expectedTotalToken0Amount);
+        int256 token1Delta = int256(currentTotalToken1Amount) - int256(expectedTotalToken1Amount);
+        int256 token1DeltaInToken0;
+
+        if (token1Delta < 0) {
+            token1DeltaInToken0 = -int256(
+                FullMath.mulDiv(uint256(-token1Delta), CommonLibrary.Q96, domainPositionParams.spotPriceX96)
+            );
+        } else {
+            token1DeltaInToken0 = int256(
+                FullMath.mulDiv(uint256(token1Delta), CommonLibrary.Q96, domainPositionParams.spotPriceX96)
+            );
+        }
+
+        int256 minDeviation = int256(
+            FullMath.mulDiv(
+                expectedTotalToken0Amount +
+                    FullMath.mulDiv(expectedTotalToken1Amount, CommonLibrary.Q96, domainPositionParams.spotPriceX96),
+                ratioParams.minRebalanceDeviationD,
+                DENOMINATOR
+            )
+        );
+        return minDeviation + token0Delta <= token1DeltaInToken0 || minDeviation + token1DeltaInToken0 <= token0Delta;
+    }
+
+    /// @notice returns true if the rebalance between assets on different vaults is needed
+    /// @param currentTokenAmounts the current amounts of tokens on the vaults
+    /// @param expectedTokenAmounts the amounts of tokens expected after rebalance
+    /// @param ratioParams ratio of the tokens between erc20 and money vault combined with needed deviations for rebalance to be called
+    /// @return needed true if the rebalance is needed
+    function tokenRebalanceNeeded(
```

Comment:
Why do we do rebalances only if a share of erc20Vault deviates from expected, but not Uni or Aave? Isn't it possible that erc20vault doesn't deviate but UNi and Aave do?